sway-core/src/ir_generation/function.rs

```diff
@@ -1354,6 +1480,437 @@ impl<'eng> FnCompiler<'eng> {
+                        let item_ptr = save_to_local_return_ptr(self, context, item);
+
+                        let offset = Constant::new_uint(context, 64, 4);
+                        let offset = Value::new_constant(context, offset);
+                        let item_ptr =
+                            calc_addr_to_ptr_u8(&mut self.current_block, context, item_ptr, offset);
+
+                        // now copy 4 bytes
+                        let addr = calc_addr_to_ptr_u64(&mut self.current_block, context, ptr, len);
+                        self.current_block
+                            .append(context)
+                            .mem_copy_bytes(addr, item_ptr, 4);
+                        increase_len(&mut self.current_block, context, len, 4)
+                    }
+                    TypeInfo::UnsignedInteger(IntegerBits::SixtyFour) => {
+                        assert!(item.get_type(context).unwrap().is_uint64(context));
+                        let addr = calc_addr_to_ptr_u64(&mut self.current_block, context, ptr, len);
+                        append_u64(&mut self.current_block, context, addr, len, item)
+                    }
+                    TypeInfo::UnsignedInteger(IntegerBits::V256) | TypeInfo::B256 => {
+                        // Save to local and return ptr to local
+                        let item_ptr = save_to_local_return_ptr(self, context, item);
+                        let addr = calc_addr_to_ptr_u8(&mut self.current_block, context, ptr, len);
+                        self.current_block
+                            .append(context)
+                            .mem_copy_bytes(addr, item_ptr, 32);
+                        increase_len(&mut self.current_block, context, len, 32)
+                    }
+                    TypeInfo::StringArray(string_len) => {
+                        // Save to local and return ptr to local
+                        let item_ptr = save_to_local_return_ptr(self, context, item);
+                        let addr = calc_addr_to_ptr_u8(&mut self.current_block, context, ptr, len);
+                        self.current_block.append(context).mem_copy_bytes(
+                            addr,
+                            item_ptr,
+                            string_len.val() as u64,
+                        );
+                        increase_len(
+                            &mut self.current_block,
+                            context,
+                            len,
+                            string_len.val() as u64,
+                        )
+                    }
+                    TypeInfo::StringSlice | TypeInfo::RawUntypedSlice => {
+                        let uint64 = Type::get_uint64(context);
+
+                        let item_ptr = save_to_local_return_ptr(self, context, item);
+                        let addr = calc_addr_to_ptr_u64(&mut self.current_block, context, ptr, len);
+
+                        let addr_ident = Ident::new_no_span("addr".into());
```

Comment:
Having a nice comment here like done in other places how the resulting `asm` block looks like would be nice. Seeing the resulting asm is very helpful.

Reply:
Done.